### PR TITLE
Add `:ios-app` and let CI upload each platform output as artifacts

### PR DESCRIPTION
**Pull request by eskatos**
The ios application shares the UI with android and all desktop platforms.
The ios app does not reuse resources generated by the build yet, one step at a time :)

CI produces a ZIP file per platform.
The ios `.app` is unsigned so it cannot be installed to a device but it can be dragged and dropped to an iphone simulator.

See e.g. https://github.com/gradle/imaginate/actions/runs/4722740997

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -32,66 +36,30 @@ jobs:
...
-        run: ./gradlew --continue --stacktrace :buildSrc:build
-        env:
-          CI: true
-
-      - name: Build Applications
-        run: ./gradlew --continue --stacktrace build
-        env:
-          CI: true
-
-      - name: Comment PR with build scan on failure
```
Comment (bamboo) [inline .github/workflows/ci.yml:61]: No more build scan on failures?

**Review by bamboo — APPROVED** [review]
🎉

File: .github/workflows/ci.yml (hunk 1 above)
Comment (eskatos) [reply .github/workflows/ci.yml:61]: All builds still publish build scans and the gradle build action summary still provides all the links.

What got removed is the automatic comment to the PR. the way it was done didn't work well with composite actions, I'll see at reinstating this another way.

Comment (bamboo) [reply .github/workflows/ci.yml:61]: I see, thanks for the explanation.

**Pull request merged**